packages/shared/lib/contexts/onboarding/actions/initialiseOnboardingProfile.ts

```diff
@@ -1,16 +1,30 @@
-import { NetworkProtocol } from '@core/network'
-import { profileManager } from '@core/profile-manager'
 import { get } from 'svelte/store'
+
+import { stopPollingLedgerNanoStatus } from '@core/ledger'
+import { NetworkProtocol } from '@core/network'
+import { destroyProfileManager, profileManager } from '@core/profile-manager'
+
 import { OnboardingProfileManagerAlreadyInitializedError } from '../errors'
 import { buildOnboardingProfile } from '../helpers'
-import { onboardingProfile } from '../stores'
+import { isOnboardingLedgerProfile, onboardingProfile } from '../stores'
 
 /**
  * Builds a new onboarding profile and sets the Svelte store accordingly.
  */
-export function initialiseOnboardingProfile(isDeveloperProfile: boolean, networkProtocol?: NetworkProtocol): void {
+export function initialiseOnboardingProfile(
+    isDeveloperProfile: boolean,
+    networkProtocol?: NetworkProtocol,
+    overrideExistingProfileManager = false
+): void {
     if (get(profileManager)) {
-        throw new OnboardingProfileManagerAlreadyInitializedError()
+        if (overrideExistingProfileManager) {
```

Comment:
Should this clearing down not be done in a clear onboarding function called in intialiseOnboarding